src/UseCases/AddSubscription/AddSubscriptionUseCase.php

```diff
@@ -45,7 +46,12 @@ public function addSubscription( SubscriptionRequest $subscriptionRequest ) {
 		$this->subscriptionRepository->storeSubscription( $subscription );
 
 		$postalAddress = $subscription->getAddress();
-		$this->message->setTemplateParams( [ 'subscription' => $subscription ] );
+		$confirmationCodeConverter = new ConfirmationCodeConverter();
+		$confirmationCode = $confirmationCodeConverter->fromBinaryToReadable( $subscription->getConfirmationCode() );
```

Comment:
Like this details from the database schema leak out from the data access layer, skip the domain layer and end up in the application layer. It'd be much nicer if this was taken care of by the persistence service.